### PR TITLE
[release-1.57] CVE-2023-45288: Bump golang.org/x/net to v0.23.0

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

Address CVE-2023-45288 by bumping golang.org/x/net to v0.23.0.

Though we don't strictly use golang.org/x/net to create our servers, other repositories are playing safe with their dependencies and bumping the library.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://nvd.nist.gov/vuln/detail/CVE-2023-45288

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mhenriks** [comment]
/hold

use `require` not replace

File: go.mod [hunk 1]
```diff
@@ -128,13 +128,13 @@ require (
 	go.opencensus.io v0.24.0 // indirect
 	go.uber.org/atomic v1.7.0 // indirect
 	go.uber.org/multierr v1.6.0 // indirect
-	golang.org/x/crypto v0.14.0 // indirect
+	golang.org/x/crypto v0.21.0 // indirect
 	golang.org/x/exp v0.0.0-20230321023759-10a507213a29 // indirect
 	golang.org/x/mod v0.9.0 // indirect
-	golang.org/x/net v0.17.0 // indirect
+	golang.org/x/net v0.23.0 // indirect
```
Comment (akalenyu) [inline go.mod:134]: doesn't this get whacked when the next developer pulls in a library and runs dep update?

**Comment by awels** [comment]
it looks in the require section, not replace section.

**Pull request merged**